Google.PowerShell/Common/AuthenticateWithSdkCredentialsExecutor.cs

```diff
@@ -102,8 +75,7 @@ public AuthenticateWithSdkCredentialsExecutor()
         {
             if (args.Response.StatusCode == HttpStatusCode.Unauthorized)
             {
-                return !Object.Equals(Token.AccessToken, flow.AccessMethod.GetAccessToken(args.Request))
-                    || await RefreshTokenAsync(args.CancellationToken).ConfigureAwait(false);
+                return await RefreshTokenAsync(args.CancellationToken).ConfigureAwait(false);
```

Comment:
What is the `IHttpUnsuccessfulResponseHandler` interface for and what is the significance of  `HandleResponseAsync` 's return value?

If we successfully refresh the auth token, does it retry the request? What if you legitimately access a resource you don't have access to?

We should probably add a comment to the top of the method explaining its role, even if it is just a copy of the interface's documentation. Do you agree?